### PR TITLE
ci: pick up cilium-cli v0.11.9 for master/v1.11 workflows

### PR DESCRIPTION
**Pull request by tklauser**
The --base-version flag is no longer necessary and has been deprecated,
see https://github.com/cilium/cilium-cli/pull/891 for details.

Release notes: https://github.com/cilium/cilium-cli/releases/tag/v0.11.9

### CODE REVIEW

**Comment by tklauser** [comment]
Enabled CI 3.0 tests passed (all GKE-based workflows are curently disabled due to #20013, AKS currently blocked on #19379):

- AWS CNI: https://github.com/cilium/cilium/runs/6934599941?check_suite_focus=true
- EKS: https://github.com/cilium/cilium/runs/6934599465?check_suite_focus=true

Removing `DO NOT MERGE` commit for now and marking as ready for review. Not sure whether we should wait for above issue/PR to be fixed/merged so we could run other workflows as well.

**Review by sayboras — APPROVED** [review]
:shipit:

**Comment by tklauser** [comment]
Another green run of AWS-CNI, EKS and GKE:

* https://github.com/cilium/cilium/actions/runs/2527689856
* https://github.com/cilium/cilium/actions/runs/2527689839
* https://github.com/cilium/cilium/actions/runs/2527689875

Force-pushing again to also test external workloads and multicluster.

File: .github/workflows/conformance-eks-v1.11.yaml [hunk 1]
```diff
@@ -259,6 +256,8 @@ jobs:
       - name: Enable Relay
         run: |
           cilium hubble enable ${{ steps.vars.outputs.hubble_enable_defaults }}
+          # NB: necessary to work against occassional flakes due to https://github.com/cilium/cilium-cli/issues/918
+          cilium status --wait
```
Comment (pchaigno) [inline .github/workflows/conformance-eks-v1.11.yaml:260]: Nit: That would be best placed in a separate commit so we can just revert when the time comes (my understanding is that it's just a workaround).

File: .github/workflows/conformance-aws-cni-v1.10.yaml [hunk 2]
```diff
@@ -66,7 +66,7 @@ env:
   cilium_base_version: "1.10"
   cilium_cli_version: v0.10.7
   check_url: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
-  eksctl_version: v0.94.0
+  eksctl_version: v0.101.0
```
Comment (pchaigno) [inline .github/workflows/conformance-aws-cni-v1.10.yaml:69]: Do we also need to update the docs?

File: .github/workflows/conformance-aws-cni.yaml [hunk 3]
```diff
@@ -250,7 +250,7 @@ jobs:
 
       - name: Update AWS VPC CNI plugin
         run: |
-          kubectl apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.7.10/config/v1.7/aws-k8s-cni.yaml
+          kubectl apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/release-1.11/config/master/aws-k8s-cni.yaml
```
Comment (pchaigno) [inline .github/workflows/conformance-aws-cni.yaml:253]: Isn't this happening because of the previous commit updating eksctl?

**Review by pchaigno — APPROVED** [review]

File: .github/workflows/conformance-eks-v1.11.yaml (hunk 1 above)
Comment (tklauser) [reply .github/workflows/conformance-eks-v1.11.yaml:260]: Good point, will move to a separate commit.

File: .github/workflows/conformance-aws-cni.yaml (hunk 3 above)
Comment (tklauser) [reply .github/workflows/conformance-aws-cni.yaml:253]: Likely, yes. I'll squash the two commits.

File: .github/workflows/conformance-aws-cni-v1.10.yaml (hunk 2 above)
Comment (tklauser) [reply .github/workflows/conformance-aws-cni-v1.10.yaml:69]: We currently don't seem to suggest a particular `eksctl` version in the docs. Not sure whether we should?

**Comment by tklauser** [comment]
Tests are green again:

* AWS-CNI: https://github.com/cilium/cilium/actions/runs/2534564618
* EKS: https://github.com/cilium/cilium/actions/runs/2534564627
* GKE: https://github.com/cilium/cilium/actions/runs/2534564622
* Multicluster: https://github.com/cilium/cilium/actions/runs/2534564611
* External workloads: https://github.com/cilium/cilium/actions/runs/2534564620

Removing `DO NOT MERGE` commit and marking as ready to merge.

File: .github/workflows/conformance-aws-cni-v1.10.yaml (hunk 2 above)
Comment (pchaigno) [reply .github/workflows/conformance-aws-cni-v1.10.yaml:69]: I'm not sure we had a strict requirement before. It seems like now we do.

Comment (tklauser) [reply .github/workflows/conformance-aws-cni-v1.10.yaml:69]: > Do we also need to update the docs?

Addressed in #20329
